File: 345. Reverse Vowels of a String/Java/Reverse Vowels of a String.java
public class Solution {
    public String reverseVowels(String s) {
        int i = 0;
        int j = s.length() - 1;
        Set<Character> vowels = new HashSet<>();
        vowels.add('a');
        vowels.add('e');
        vowels.add('i');
        vowels.add('o');
        vowels.add('u');
        vowels.add('A');
        vowels.add('E');
        vowels.add('I');
        vowels.add('O');
        vowels.add('U');
        char[] str = s.toCharArray();
        while (i < j) {
            if (!vowels.contains(str[i])) {
                i++;
            } else if (!vowels.contains(str[j])) {
                j--;
            } else {
                if (str[i] != str[j]) {
                    char temp = str[i];
                    str[i] = str[j];
                    str[j] = temp;
                }
                i++;
                j--;
            }
        }
        return new String(str);
    }
}

File: 345. Reverse Vowels of a String/C++/Reverse Vowels of a String.cpp
class Solution {
public:
    string reverseVowels(string s) {
        int i = 0;
        int j = s.size() - 1;
        set<char> vowels;
        vowels.insert('a');
        vowels.insert('e');
        vowels.insert('i');
        vowels.insert('o');
        vowels.insert('u');
        vowels.insert('A');
        vowels.insert('E');
        vowels.insert('I');
        vowels.insert('O');
        vowels.insert('U');
        while (i < j) {
            if (vowels.find(s[i]) == vowels.end()) {
                i++;
            } else if (vowels.find(s[j]) == vowels.end()) {
                j--;
            } else {
                if (s[i] != s[j]) {
                    char temp = s[i];
                    s[i] = s[j];
                    s[j] = temp;
                }
                i++;
                j--;
            }
        }
        return s;
    }
};